File: src/main/java/org/ldk/structs/ProcessMessagesCallback.java
package org.ldk.structs;

import org.ldk.impl.bindings;
import org.ldk.enums.*;
import org.ldk.util.*;
import java.util.Arrays;
import java.lang.ref.Reference;
import javax.annotation.Nullable;

/**
 * A callback which will be called to trigger network message processing.
 * 
 * Usually, this should call [`PeerManager::process_events`].
 * 
 * [`PeerManager::process_events`]: lightning::ln::peer_handler::PeerManager::process_events
 */
@SuppressWarnings("unchecked") // We correctly assign various generic arrays
public class ProcessMessagesCallback extends CommonBase {
	final bindings.LDKProcessMessagesCallback bindings_instance;
	ProcessMessagesCallback(Object _dummy, long ptr) { super(ptr); bindings_instance = null; }
	private ProcessMessagesCallback(bindings.LDKProcessMessagesCallback arg) {
		super(bindings.LDKProcessMessagesCallback_new(arg));
		this.ptrs_to.add(arg);
		this.bindings_instance = arg;
	}
	@Override @SuppressWarnings("deprecation")
	protected void finalize() throws Throwable {
		if (ptr != 0) { bindings.ProcessMessagesCallback_free(ptr); } super.finalize();
	}
	/**
	 * Destroys the object, freeing associated resources. After this call, any access
	 * to this object may result in a SEGFAULT or worse.
	 *
	 * You should generally NEVER call this method. You should let the garbage collector
	 * do this for you when it finalizes objects. However, it may be useful for types
	 * which represent locks and should be closed immediately to avoid holding locks
	 * until the GC runs.
	 */
	public void destroy() {
		if (ptr != 0) { bindings.ProcessMessagesCallback_free(ptr); }
		ptr = 0;
	}
	public static interface ProcessMessagesCallbackInterface {
		/**
		 * The method which is called.
		 */
		void call();
	}
	private static class LDKProcessMessagesCallbackHolder { ProcessMessagesCallback held; }
	public static ProcessMessagesCallback new_impl(ProcessMessagesCallbackInterface arg) {
		final LDKProcessMessagesCallbackHolder impl_holder = new LDKProcessMessagesCallbackHolder();
		impl_holder.held = new ProcessMessagesCallback(new bindings.LDKProcessMessagesCallback() {
			@Override public void call() {
				arg.call();
				Reference.reachabilityFence(arg);
			}
		});
		return impl_holder.held;
	}
	/**
	 * The method which is called.
	 */
	public void call() {
		bindings.ProcessMessagesCallback_call(this.ptr);
		Reference.reachabilityFence(this);
	}

}

File: c_sharp/src/org/ldk/structs/ProcessMessagesCallback.cs

using org.ldk.impl;
using org.ldk.enums;
using org.ldk.util;
using System;

namespace org { namespace ldk { namespace structs {



/** An implementation of ProcessMessagesCallback */
public interface ProcessMessagesCallbackInterface {
	/**The method which is called.
	 */
	void call();
}

/**
 * A callback which will be called to trigger network message processing.
 * 
 * Usually, this should call [`PeerManager::process_events`].
 * 
 * [`PeerManager::process_events`]: lightning::ln::peer_handler::PeerManager::process_events
 */
public class ProcessMessagesCallback : CommonBase {
	internal bindings.LDKProcessMessagesCallback bindings_instance;
	internal long instance_idx;

	internal ProcessMessagesCallback(object _dummy, long ptr) : base(ptr) { bindings_instance = null; }
	~ProcessMessagesCallback() {
		if (ptr != 0) { bindings.ProcessMessagesCallback_free(ptr); }
	}

	private class LDKProcessMessagesCallbackHolder { internal ProcessMessagesCallback held; }
	private class LDKProcessMessagesCallbackImpl : bindings.LDKProcessMessagesCallback {
		internal LDKProcessMessagesCallbackImpl(ProcessMessagesCallbackInterface arg, LDKProcessMessagesCallbackHolder impl_holder) { this.arg = arg; this.impl_holder = impl_holder; }
		private ProcessMessagesCallbackInterface arg;
		private LDKProcessMessagesCallbackHolder impl_holder;
		public void call() {
			arg.call();
				GC.KeepAlive(arg);
		}
	}

	/** Creates a new instance of ProcessMessagesCallback from a given implementation */
	public static ProcessMessagesCallback new_impl(ProcessMessagesCallbackInterface arg) {
		LDKProcessMessagesCallbackHolder impl_holder = new LDKProcessMessagesCallbackHolder();
		LDKProcessMessagesCallbackImpl impl = new LDKProcessMessagesCallbackImpl(arg, impl_holder);
		long[] ptr_idx = bindings.LDKProcessMessagesCallback_new(impl);

		impl_holder.held = new ProcessMessagesCallback(null, ptr_idx[0]);
		impl_holder.held.instance_idx = ptr_idx[1];
		impl_holder.held.bindings_instance = impl;
		return impl_holder.held;
	}

	/**
	 * The method which is called.
	 */
	public void call() {
		bindings.ProcessMessagesCallback_call(this.ptr);
		GC.KeepAlive(this);
	}

}
} } }
